src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -264,20 +263,36 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 		return exprStmt.WithExpression( result );
 	}
 
-	private InvocationExpressionSyntax Transform( InvocationExpressionSyntax invocationExpr) {
+	private ExpressionSyntax Transform( InvocationExpressionSyntax invocationExpr) {
+		ExpressionSyntax newExpr = invocationExpr;
 		var memberAccess = invocationExpr.Expression as MemberAccessExpressionSyntax;
+
 		if( string.Equals( memberAccess?.Name?.Identifier.ValueText, "ConfigureAwait", StringComparison.Ordinal ) ) {
 			if( memberAccess?.Expression is not null ) {
-				invocationExpr = (InvocationExpressionSyntax)memberAccess.Expression;
+				newExpr = memberAccess.Expression;
+				return Transform( newExpr );
 			}
 		}
 		return invocationExpr
 			.WithExpression( Transform( invocationExpr.Expression ) )
 			.WithArgumentList( TransformAll( invocationExpr.ArgumentList, Transform ) );
 	}
 
-	private MemberAccessExpressionSyntax Transform( MemberAccessExpressionSyntax memberAccessExpr ) {
+	bool ShouldRemoveReturnedMemberAccess( MemberAccessExpressionSyntax memberAccessExpr )
+	  => memberAccessExpr.Name.Identifier.ValueText switch {
+		  "FromResult" => true,
+		  "CompletedTask" => true,
+		  _ => false
+	  };
+
+	private ExpressionSyntax Transform( MemberAccessExpressionSyntax memberAccessExpr ) {
 		if( memberAccessExpr.IsKind( SyntaxKind.SimpleMemberAccessExpression ) ) {
+			if( ShouldRemoveReturnedMemberAccess( memberAccessExpr ) &&
+				( memberAccessExpr.Parent.IsKind( SyntaxKind.ReturnStatement ) ||
+				( memberAccessExpr.Parent?.Parent?.IsKind( SyntaxKind.ReturnStatement ) ?? false ) ) ) {

```

Comment:
Fine for now, but I think we could probably pass around that we're in a return statement on the stack into MaybeTransform/Transform from the `ReturnStatementSyntax` arm in `private StatementSyntax Transform( StatementSyntax stmt )` rather than inspecting upwards.